nodes/test/test_ssrf_protection.py

```
@@ -0,0 +1,304 @@
+    @pytest.mark.parametrize(
+        'ip',
+        [
+            '::1',  # loopback
+            'fc00::1',  # unique local
+            'fd12:3456::1',  # unique local
+            'fe80::1',  # link-local
+        ],
+    )
+    def test_blocked_ipv6(self, ip):
+        with patch('ssrf_protection.socket.getaddrinfo', _fake_getaddrinfo(ip)):
+            with pytest.raises(SSRFError, match='private/reserved range'):
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)